# Conserved_Stromatolite_bacteria_manuscript
A collection of raw data, Python script and R scripts used to create data and images for manuscript entitled "Conserved bacterial genomes from two geographically distinct peritidal stromatolite formations shed light on potential functional guilds"

Figure 2 creation:

1. ORFs were called in Bacteria.filtered.fasta files (generated by Automneta) for all samples (CSU, SU, SM1, SM2, SL, CRU, RU, RM1, RM2, RL) using prodigal v2.6.3.
2. ORFs were assigned KO annotations using kofamscan 1.0.0
3. KO annotations for each sample were parsed using autometa_bacterial_files_kegg_parser.py. This script uses the contig coverage in the prodigal output to calculate the coverage_corrected abundance of each KO gene being queried (See Phosphate_query for an example). 
4. KO abundance was normalized against the sum of all contig coverage values per samples i.e. 

KO_abundance = coverage_corrected abundance of KO / sum (contig1_cov, contig2_cov ... contigN_cov)*100.
e.g. 

Heatmap created in R using code found in "R_code_for_heatmaps" in this repo.


Figure 5 creation:

Heatmap in the manuscript were created in iTol. Data was prepared in the following manner:

1. ORFs were called in all genomes using prokka 1.14.5.
2. ORFs were assigned KO annotations using kofamscan 1.0.0
3. KO annotations for each bin (in Kofamscan output folder) were parsed using kegg_parser.py to obtained counts of all KO entries listed in Phosphate_query and other metabolite ther query files. Phosphate_query and other query files used in this study were created manually using a combination of literature and KEGG searches.
4. KO counts were normalized to bin coverage i.e KO abundance = KO count*(bin_coverage/sum_of_all_Bins_coverage_in_sample)*100.
See Raw_KO_counts_with_metadata.xlsx for clarification
